webapp/components/test-search.html

```diff
@@ -86,50 +86,36 @@
         this.onBlur = this.handleBlur.bind(this);
       }
 
-      computeAutocompleteURL(testRuns, q) {
-        if (!testRuns || !testRuns.length) {
-          return '';
-        }
-        let url = new URL('/api/autocomplete', window.location);
-        url.searchParams.set('limit', 10);
-        url.searchParams.set(
-          'run_ids',
-          testRuns.map(r => r.id.toString()).join(','));
-        if (q) {
-          url.searchParams.set('q', q);
-        }
-        return url;
-      }
-
       ready() {
         super.ready();
-        this.q = this.query;
+        this._createMethodObserver('pathsUpdated(query, testPaths)');
+        this.queryInput = this.query;
       }
 
-      async observeAutocompleteURL(newValue, oldValue) {
-        if (newValue === oldValue || !newValue) {
-          return;
-        }
+      queryUpdated(query) {
+        this.queryInput = query;
+      }
 
-        const response = await window.fetch(newValue);
-        if (response.status !== 200) {
-          throw response;
+      pathsUpdated(query, paths) {
```

Comment:
nit: This depends on query too, and what it _really_ does is update the datalist. Maybe name it `updateDatalist` or similar?

Reply:
Done.